frontend/src/utils/dataFetching/fetchPosts.js

```diff
@@ -0,0 +1,41 @@
+import { getToken } from "../auth";
+
+// FETCH ALL POSTS
+export const fetchData = async url => {
+  let optionsToken = null;
+
+  if (getToken()) {
+    optionsToken = {
+      headers: {
+        Authorization: `Bearer ${getToken()}`,
+        "Content-Type": "application/json"
+      }
+    };
+  }
+
+  const response = await fetch(url, optionsToken);
+  const data = response.json();
+
+  return data;
+};
+
+// FETCH SINGLE POST
+export const fetchSinglePost = async (url, id) => {
+  const response = await fetch(url + id);
+  const post = response.json();
+
+  return post;
+};
+
+// FETCH USERS POSTS
+export const fetchUserPosts = async (url, id) => {
+  try {
+    const response = await fetch(url + id);
+    const user = await response.json();
+    const userPosts = user.posts;
+
+    return userPosts;
+  } catch (error) {
+    console.warn(error);
+  }
+};
```

Comment:
Hmm... why fetch posts is in other scope than post crud? CRUD stands for Create Read - which is fetch, update and delete.

Reply:
makes perfect sense